js/src/oai.ts

```diff
@@ -7,19 +7,24 @@ import { openAIV4ProxyWrapper } from "./oai_wrappers/proxy_wrappers";
  * Currently, this only supports the `v4` API.
  *
  * @param openai The `OpenAI` object.
- * @param proxyBaseUrl Use the Braintrust proxy (https://github.com/braintrustdata/braintrust-proxy) as the base URL. When left `undefined`, the URL is obtained from the environment variable `BRAINTRUST_PROXY_URL`, defaulting to `https://braintrustproxy.com/v[api_version]`. Pass `null` to leave the original base URL intact and use the non-proxy wrapper, or pass a string to use a custom URL.
- * @param proxyApiKey For clients using the proxy wrapper (`proxyBaseUrl` is not `null`). When left `undefined`, the API key is set from `BRAINTRUST_API_KEY` if available. Pass `null` to leave the original API key intact, or pass a string to use a custom API key.
+ * @param options.useProxy By default (or if `false`), the wrapper does not trace through the proxy. Pass `true` to use the Braintrust proxy as the base URL. The URL is obtained from the environment variable `BRAINTRUST_PROXY_URL`, defaulting to `https://braintrustproxy.com/v1. Pass a string to use a custom proxy URL.
+ * @param options.apiKey Only used when `useProxy` is set. By default, the API key is set from `BRAINTRUST_API_KEY` if available. Pass a string to use a custom API key.
```

Comment:
I don't quite follow the `apiKey` setting. Why would someone need to set this, instead of setting the api key in the OpenAI client? I'm imagining the code like

```
wrapOpenAI(new OpenAI({ apiKey: process.env.BRAINTRUST_API_KEY }), { useProxy: true })
```

Reply:
I guess there are two ways someone can initialize the OpenAI client
```
// Uses OPENAI_API_KEY
new OpenAI();

// Uses custom api key
new OpenAI({apiKey: [custom_api_key]});
```

I was thinking that in the first case (which I imagine is the most common), they don't have to change anything about their code (just add the wrapper) and it'll change the API key to `BRAINTRUST_API_KEY`. In the second case, they can move the `apiKey` argument to the wrapper to tell it to set a custom API key instead of `BRAINTRUST_API_KEY`. But that wouldn't work if they hadn't defined `OPENAI_API_KEY` at all.

We could change the API key functionality so that if `openai.apiKey === process.env.OPENAI_API_KEY`, only then do we change it to `BRAINTRUST_API_KEY`, otherwise we leave it intact? And we could add a flag `keepApiKey?: boolean`, which if true, will not alter the API key at all?